snap/hooks/configure

```diff
@@ -0,0 +1,11 @@
+#! /bin/bash
+
+config_file="$SNAP_DATA/snap-config.yaml"
+provider="$(snapctl get provider)"
+
+cat <<EOT > "${config_file}"
```

Comment:
instead of writing to a file why don't we call get in the relevant locations?

Reply:
Because `snap get` requires root access

Reply:
Details for posterity:
### snap get
- always requires root access
### snapctl get
- doesn't require root access when executed inside a snap
- fails when executed outside a snap (like during a unit test...)